### PR TITLE
Allow for random IO workloads via fio - default to sequential

### PR DESCRIPTION
**Pull request by joehandzik**
Signed-Off-By: Joe Handzik <jhandzik@nvidia.com>

### CODE REVIEW

File: bobber/lib/tests/run_tests.py [hunk 1]
```diff
@@ -83,13 +83,17 @@ def run_stg_bw(args: Namespace, bobber_version: str, iteration: int,
                               f'threads_{args.bw_threads}_'
                               f'direct_{args.direct}_'
                               f'depth_{args.io_depth}_'
+                              f'read_pattern_{args.read_pattern}_'
+                              f'write_pattern_{args.read_pattern}_'
```
Comment (roclark) [inline bobber/lib/tests/run_tests.py]: This doesn't change anything with our parser, correct? I don't think it should as it shouldn't expect a rigid structure for the log's filename, but something I'd want to double check.

File: bobber/bobber.py [hunk 2]
```diff
@@ -131,6 +133,16 @@ def parse_args(version: str) -> Namespace:
...
+    commands_parent.add_argument('--read-pattern', help='Specify IO pattern '
+                                 'for fio read tests. Supported values: '
+                                 'sequential, random. Defaults to sequential.',
+                                 default='sequential',
+                                 choices=READ_PATTERNS.keys())
+    commands_parent.add_argument('--write-pattern', help='Specify IO pattern '
+                                 'for fio write tests. Supported values: '
+                                 'sequential, random. Defaults to sequential.',
+                                 default='sequential',
+                                 choices=WRITE_PATTERNS.keys())
```
Comment (roclark) [inline bobber/bobber.py]: Looks like there's some extra whitespace at the end of this line.

**Review by roclark — APPROVED** [review]
Awesome stuff! Looks good once the whitespace is removed.

File: bobber/bobber.py (hunk 2 above)
Comment (joehandzik) [reply bobber/bobber.py]: Yeah I cleaned that up...was weirdly misbehaving (or not behaving as I expected, anyway) in VS Code.

File: bobber/lib/tests/run_tests.py (hunk 1 above)
Comment (joehandzik) [reply bobber/lib/tests/run_tests.py]: Still need to test - worst case, will try to test locally on my Mac.

Comment (roclark) [reply bobber/lib/tests/run_tests.py]: Let me know if you want me to run on my workstation and I can test it out too.

**Comment by joehandzik** [comment]
Updated read-pattern argument to random in the CI test - this lets us test the default and the alternate argument between reads and writes in the same test sweep. If this passes CI, I am comfortable merging it.

**Comment by joehandzik** [comment]
@roclark - CI job shows the expected IO sequential/random IO patterns and completes fine. Tracking #80 for a further enhancement to make it super clear what the IO pattern was in result reporting. Merging!

**Pull request merged**